Lib/fontTools/feaLib/builder.py

```diff
@@ -914,10 +914,19 @@ def add_single_subst(self, location, prefix, suffix, mapping, forceChain):
         lookup = self.get_lookup_(location, SingleSubstBuilder)
         for (from_glyph, to_glyph) in mapping.items():
             if from_glyph in lookup.mapping:
-                raise FeatureLibError(
-                    'Already defined rule for replacing glyph "%s" by "%s"' %
-                    (from_glyph, lookup.mapping[from_glyph]),
-                    location)
+                if to_glyph == lookup.mapping[from_glyph]:
+                    # log warning?
+                    # FDK logs "[NOTE] Removing duplicate single substitution in 'xxxx' feature: from_glyph, to_glyph" in this case
+                    log.info(FeatureLibError(
```

Comment:
i believe you're using the FeatureLibError exception so that it logs the line number, right?
if so make it more explicit, e.g. turn the exception into a string first and pass a string to log.info, instead of the exception object.